framework/Source/Sparkles.swift

```diff
@@ -197,4 +212,31 @@ extension Sparkles {
             --> output
         }
     }
+
+    private func reconnectFaceMaskInput(_ maskInput: PictureInput) {
+        let semaphore = DispatchSemaphore(value: 0)
+
+        maskInput.addTarget(lightExtractorEffect, atTargetIndex: 2)
+        Task {
+            await maskInput.processImage()
+            semaphore.signal()
+        }
+        semaphore.wait()
+    }
+}
+
+private extension UIImage {
+    static func makeEmptyMaskImage() -> UIImage {
+        let size = CGSize(width: 1, height: 1)
+        let color = UIColor.clear
```

Comment:
question: So, an empty mask is a clear color? The mask should be black and white, right?

The old implementation was black.

Reply:
Yes the empty mask is a clear color; the old implementation used the r channel as the mask, but we decided to change to use the alpha channel at last.

Reply:
Wait... you meant the previous implementation in this PR
Sorry that was just my mistake 🙏🏻
I remember it wrong with the first version of the face mask